feature/src/main/java/com/teamdontbe/feature/login/LoginViewModel.kt

```diff
@@ -0,0 +1,28 @@
+package com.teamdontbe.feature.login
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.teamdontbe.core_ui.view.UiState
+import com.teamdontbe.domain.entity.AuthEntity
+import com.teamdontbe.domain.repository.AuthRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val authRepository: AuthRepository
+) : ViewModel() {
+    private val _postLogin = MutableStateFlow<UiState<AuthEntity?>>(UiState.Empty)
+    val postLogin: StateFlow<UiState<AuthEntity?>> = _postLogin
+
+    fun login(auth: String, socialType: String) = viewModelScope.launch {
+        authRepository.login(auth, socialType).collectLatest {
+            _postLogin.value = UiState.Success(it)
```

Comment:
```suggestion
 if (it != null) {
  _postLogin.value = UiState.Success(it)
 }else{
   _postLogin.value = UiState.Empty(it)
 }
           
```
p1 : base response의 data를 nullable로 가져가고자 합니다! 
data가 non-null인 것도 있어서 base response를 두개로 만들려고 했으나, 나중에 dto가 nullable로 변할 수도 있어서(변경 사항에 따라) 미리 null 대응을 해두면 좋을 것 같아요!

따라서 null 일 때도 분기처리 따로 해주면 좋을 것 같아요-!

Reply:
수정 완~